utils/ghc-toolchain/src/GHC/Toolchain/Tools/MergeObjs.hs

```diff
@@ -22,7 +22,7 @@ data MergeObjs = MergeObjs { mergeObjsProgram :: Program
 
 findMergeObjs :: ProgOpt -> Cc -> CcLink -> Nm -> M MergeObjs
 findMergeObjs progOpt cc ccLink nm = checking "for linker for merging objects" $ do
-    prog <- findProgram "linker for merging objects" progOpt ["ld.gold", "ld"]
+    prog <- findProgram "linker for merging objects" progOpt ["ld"]
```

Comment:
This should ultimately also end up being a seperate PR for upstreaming.

Reply:
https://gitlab.haskell.org/ghc/ghc/-/merge_requests/14942